Automattic/woocommerce-payments
# Show dispute details on the transaction details screen when a dispute is awaiting response and past due

Fixes #7229

> **Note**
> This PR is behind feature flag `_wcpay_feature_dispute_on_transaction_page`


#### Changes proposed in this Pull Request

This PR allows the Transaction Details dispute details to be rendered if a dispute is awaiting a response but is past its `due_by` date.

It is uncertain whether this could occur for accounts in live mode since a dispute will move to a `won`/`lost`/`under_review` status once the `due_by` date has passed.

This case does occur when in test mode since disputes created in test mode remain as `needs_reponse`/`warning_needs_response` status until explicit action is taken (challenge or accept), even if it is past the `due_by` date.


#### Screenshots

**Current Transaction Details screen** when a dispute is awaiting a response, `due_by` in the past:
![image](https://github.com/Automattic/woocommerce-payments/assets/3147296/7906e46d-6e1b-4bd7-a5db-de682ae7c28e)

**Suggested Transaction Details screen** when a dispute is awaiting a response, `due_by` in the past:
![image](https://github.com/Automattic/woocommerce-payments/assets/3147296/97a86950-dc90-48e8-9be3-384a29298b0d)

Note the **(Past due)** text.

#### Testing instructions






* Enable feature flag by running `update_option( '_wcpay_feature_dispute_on_transaction_page', '1' );` in [WP Console](https://wordpress.org/plugins/wp-console/) or by modifying your database table `wp_options` directly.
* Create a disputed transaction using the card `4000000000000259` at checkout.
* Wait 7+ days ⚠️ 
* Find the dispute in Payments → Disputes and click it to view the dispute details screen.
* Click on the Transaction ID to view the Transaction Details screen.
* On `trunk`, you will not see any dispute details, even though the dispute's status is `Needs response`.
* On this PR, you should see the dispute details.

-------------------


**Post merge**
